File: server/CSM Gate Foundation Server/CSM Gate Foundation Core/Services/AuthService.cs
using CSM_Gate_Foundation_Core.Core.Models;
using CSM_Gate_Foundation_Core.Managers.Abstractions.Interfaces;
using CSM_Gate_Foundation_Core.Services.Abstractions.Interfaces;
using CSM_Gate_Foundation_Core.Services.Models.Inputs;

namespace CSM_Gate_Foundation_Core.Services;

/// <inheritdoc cref="IAuthService"/>
public class AuthService
    : IAuthService {

    /// <summary>
    ///     Manager for session handling and context.
    /// </summary>
    readonly ISessionsManager _sessionsManager;

    /// <summary>
    ///     Creates a new instance.
    /// </summary>
    /// <param name="sessionManager"></param>
    public AuthService(ISessionsManager sessionManager) {
        _sessionsManager = sessionManager;
    }

    /// <inheritdoc/>
    public async Task<SessionData> Authenticate(AuthInput input) {
        return await _sessionsManager.Auth(input);
    }
}

File: client/csm_gate_foundation_client/lib/src/services/auth_service.dart
import 'dart:core' hide Uri;
import 'package:csm_client_core/csm_client_core.dart';
import 'package:csm_gate_foundation_client/src/abstractions/bases/gate_foundation_service_base.dart';
import 'package:csm_gate_foundation_client/src/core/models/session_data.dart';
import 'package:csm_gate_foundation_client/src/gate_foundation_server_resolver.dart';
import 'package:csm_gate_foundation_client/src/models/inputs/auth_input.dart';
import 'package:csm_gate_foundation_client/src/services/abstractions/interfaces/iauth_service.dart';

/// Represents a server service communication handler for {Auth} operations.
class AuthService extends GateFoundationServiceBase implements IAuthService {
  /// Creates a new instance.
  AuthService(
    Uri host, {
    String? servicePath,
    super.client,
    super.headers,
  }) : super(
         host,
         servicePath ?? "auth",
       );

  @override
  Future<GateFoundationServerResolver<SessionData>> authenticate(AuthInput input) async {
    final IResponseController controller = await postSecure('authenticate', input);
    return GateFoundationServerResolver<SessionData>(controller);
  }
}
